### ISSUE TITLE
Existing IAM Role for ApigatewayToStepFunctionsRole 

### ISSUE THREAD

**Issue by dtabolich** [horike37/serverless-step-functions#91]
# This is a Feature Proposal
It would be great to have an option to override ApigatewayToStepFunctionsRole role with already existing one.

## Description
Due to our policy, it's not allowed to CreateRole() in IAM during deployment, but instead it should be provisioned via special process. So, it would be great to have an option to bypass ARN of existing role to APIG.

For feature proposals:
* This should improve flexibility and cover use cases when IAM is managed outside of deployment flow.

PS. Probably, this is something what could be done via customisation of generated CloudFormation template in Resources section. But, I haven't found how to achieve that.

Thanks.

**Comment by horike37** [comment]
Thank you for the feature request @dtabolich  :+1:
Looks a good idea, it would be good to introduce the functionality.
Do you have any ideas how is setting in serverless.yml something like to do that?

**Comment by Yoga2Pro** [comment]
Second this. In the same situation that company does not allow creating IAM roles, but only assume existing IAM role.
Thanks in advance!

**Comment by dtabolich** [comment]
Hey, sorry for being out of touch.
The request is still valid, and the way I see it could be done is via specification of 'assumeRole' on the apiGateway configuration level.
Something like:

provider:
  name: aws
  apiGateway:
    ...
     assumeRole: _arn:ofMyRole_
 
This might require conditional check somewhere here, so we can embed existing IAM Role vs. compiling inline one:
https://github.com/horike37/serverless-step-functions/blob/b08a3e282d25b3add2fa0bff5fc61edbd6a1f1d2/lib/deploy/events/apiGateway/iamRole.js#L7

Thanks.

**Comment by mraerino** [comment]
Is anyone working on this?
I'd probably take this on soon if nobody else is.

**Comment by ksteigerwald** [comment]
@Yoga2Pro Our company does not allow for IAMRole creation. If a role is specified under the provider key, I think it should be assumed that is the dedicated role for the entire process.

**Comment by theburningmonk** [comment]
How about something like this:

```yml
stepFunctions:
  stateMachines:
    hello:
      events:
        - http:
            path: posts/create
            method: POST
            iamRole: arn:aws:iam::<accountId>:role/<roleName>
```

@horike37 @Yoga2Pro @ksteigerwald

**Comment by ksteigerwald** [comment]
@theburningmonk I am having an issue with the creation of the statemachine. APIGW config is not used, would this still apply?

**Comment by theburningmonk** [comment]
@ksteigerwald is it related to the IAM Role change I made in #208?

**Comment by theburningmonk** [comment]
@ksteigerwald how are you testing my change? I usually start a project and manually update the code in the `node_modules` folder to make sure it works as expected.

**Comment by ksteigerwald** [comment]
@theburningmonk when applied to settings.stateMachine.myFunc.definition it fails. I get the same cannot create error along with log group failures... 

CloudFormation - CREATE_FAILED - AWS::IAM::Role - IamRoleStateMachineExecution

**Comment by ksteigerwald** [comment]
No, it would be under the create role if I were to guess. #208 is exclusive to GW.

**Comment by theburningmonk** [comment]
@ksteigerwald if the error is unrelated to this issue then please open a new issue and include a minimal `serverless.yml` that I can use to reproduce the issue.

**Comment by theburningmonk** [comment]
this has been released as part of 1.23.0
